### PR TITLE
covered in docker

### PR DESCRIPTION
**Pull request by Nikostrik**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    ports:
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+            "spring.datasource.url" : "jdbc:mysql://mysqldb:$MYSQLDB_DOCKER_PORT/$MYSQL_DB?useSSL=false&allowPublicKeyRetrieval=true&createDatabaseIfNotExist=true",
+            "spring.datasource.username" : "$MYSQL_USER",
+            "spring.datasource.password" : "$MYSQL_ROOT_PASSWORD",
+            "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.MySQLDialect",
+            "spring.jpa.hibernate.ddl-auto" : "update"
+          }'
```
Comment (sokimaaa) [inline docker-compose.yml:30]: Prefer using env variables over the property overriding
```suggestion
```

File: pom.xml [hunk 2]
```diff
@@ -21,6 +21,11 @@
         <jjwt.version>0.11.5</jjwt.version>
     </properties>
     <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-docker-compose</artifactId>
+        </dependency>
```
Comment (sokimaaa) [inline pom.xml:27]: Do you need this?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,8 +1,3 @@
-spring.datasource.url=jdbc:mysql://localhost:3306/book_store_db?serverTimezone=UTC
-spring.datasource.username=root
-spring.datasource.password=124533
-spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-
 spring.jpa.hibernate.ddl-auto=validate
```
Comment (sokimaaa) [inline src/main/resources/application.properties:1]: Why are you deleting this? You will be unable to connect db after this

spring.datasource.url=jdbc:mysql://localhost:3306/book_store_db?serverTimezone=UTC
spring.datasource.username=root
spring.datasource.password=124533

File: .env [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+MYSQL_ROOT_PASSWORD=123456
+MYSQL_PASSWORD=123456
+MYSQL_DB=db
+MYSQLDB_LOCAL_PORT=3307
+MYSQLDB_DOCKER_PORT=3306
+HIBERNATE_DIALECT=org.hibernate.dialect.MySQLDialect
+HIBERNATE_DDL_AUTO=update
+
+SPRING_LOCAL_PORT=8088
+SPRING_DOCKER_PORT=8080
```
Comment (kerrrusha) [inline .env:11]: add blank line to the EOF so GitHub wont mark it with red sign

**Review by kerrrusha — APPROVED** [review]

**Pull request merged**